### PR TITLE
ALMA integration tests fix

### PR DESCRIPTION
**Pull request by andamian**
Updates to the ALMA integration tests to use ALMA production servers.

A fix for #2157

### CODE REVIEW

**Review by keflavich — APPROVED** [review]
Looks good.

Could you please add some documentation about the `--alma-site` testing to https://astroquery.readthedocs.io/en/latest/testing.html ?

**Comment by andamian** [comment]
@keflavich , @bsipocz - I think this is ready. Please have a look. Thanks

File: astroquery/alma/tests/test_alma_remote.py [hunk 1]
```diff
@@ -59,8 +54,8 @@ def fin():
         request.addfinalizer(fin)
         return my_temp_dir
 
-    def test_public(self):
-        alma = get_client()
+    def test_public(self, alma):
+        assert alma.archive_url == 'https://almascience.eso.org'
```
Comment (keflavich) [inline astroquery/alma/tests/test_alma_remote.py:58]: isn't this going to depend on where you run the tests?  Or am I missing the part where eso is the default?

**Review by keflavich — APPROVED** [review]
lgtm, just have one URL-related question

File: astroquery/alma/tests/test_alma_remote.py (hunk 1 above)
Comment (andamian) [reply astroquery/alma/tests/test_alma_remote.py]: Doh! That has no place there - just a leftover from my debugging. Thanks for spotting it.

File: astroquery/alma/tests/test_alma_remote.py [hunk 2]
```diff
@@ -29,27 +28,23 @@
                 'QA2 Status', 'Group ous id', 'Pub'}
 
 
-def get_client():
+@pytest.fixture
+def alma(request):
+    """
+    Returns an alma client class. `--alma-site` pytest option can be used
+    to have the client run against a specific site
+    :param request: pytest request fixture
```
Comment (bsipocz) [inline astroquery/alma/tests/test_alma_remote.py:36]: doesn't matter here as we don't docstring test functions, but normally we use numpydoc style docstring formatting

File: astroquery/conftest.py [hunk 3]
```diff
@@ -47,3 +47,13 @@ def pytest_configure(config):
 packagename = os.path.basename(os.path.dirname(__file__))
 TESTED_VERSIONS[packagename] = version
 TESTED_VERSIONS['astropy_helpers'] = astropy_helpers_version
+
+
+def pytest_addoption(parser):
+    parser.addoption(
```
Comment (bsipocz) [inline astroquery/conftest.py:53]: again, this is nit pick, we don't use 'black', formatting so no need to put every arg in a new line 😄

File: setup.cfg [hunk 4]
```diff
@@ -32,7 +32,7 @@ astropy_header = true
 text_file_format = rst
 xfail_strict = true
 remote_data_strict = true
-addopts = --doctest-rst
+#addopts = --doctest-rst
```
Comment (bsipocz) [inline setup.cfg:35]: Please remove this comment, we started to doctest narrative docs for a couple of modules and eventually would like to see all of them being tested

Comment (bsipocz) [inline setup.cfg:35]:
```suggestion
addopts = --doctest-rst
```

**Review by bsipocz — APPROVED** [review]
Nitpick, except that we shouldn't disable rst doctesting. I'll add that commit, and then merge.

**Comment by bsipocz** [comment]
I've also rebased to be able to add a changelog for the new pytest option.

**Comment by bsipocz** [comment]
Thanks @andamian!

**Pull request merged**